app/src/main/java/org/wikipedia/page/PageFragment.java

```diff
@@ -1239,4 +1247,21 @@ void openImageInGallery(@NonNull String language) {
         leadImagesHandler.openImageInGallery(language);
     }
 
+    private void maybeShowFundraisingDialog() {
+        if (Prefs.hasVisitedArticlePage() && Prefs.fundraisingDialogShownInYear() < Calendar.getInstance().get(Calendar.YEAR)) {
+            disposables.add(ServiceFactory.getRest(getTitle().getWikiSite()).getAnnouncements()
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(list -> {
+                        String country = GeoUtil.getGeoIPCountry();
+                        Date now = new Date();
+                        for (Announcement announcement : list.items()) {
+                            if (shouldShow(announcement, country, now) && announcement.placement().equals(FUNDRAISING_PLACEMENT_ARTICLE)) {
```

Comment:
The `FUNDRAISING_PLACEMENT_` constants should probably be renamed to `ANNOUNCEMENT_PLACEMENT_`. I can easily imagine wanting to make an announcement that doesn't have to do with fundraising.

Reply:
Makes sense. Done